### ISSUE TITLE
[Feature Request] Allow function to transform names of new pages

### ISSUE THREAD

**Issue by wbthomason** [lervag/wiki.vim#64]
# Description

It would be useful to have something like `g:wiki_link_target_map` for creating new pages - a variable that a user can set to a `FuncRef` which will be called to transform the names of new wiki pages created via `WikiOpen`/similar.

# Use case example

One example use (maybe the main use?) of this would be to append the date of creation to the filenames of new pages, transforming e.g. `my_new_page` to `my_new_page_20200411`, etc.

**Comment by jabirali** [comment]
Some questions about how something like that would work:

- Considering your example, does this mean that instead of `[[my_page]]` links you always want to write `[[my_page_20200411]]` and so on? 
- If so, do you want calling `WikiLinkOpen` on a non-existing link like `[[my_page]]` to change the source to `[[my_page_20200411]]` to be consistent with the new page? 
- What about a to a non-existing target like `[[my_page_20200411]]` which occurs when a file has been deleted, that would in that case change the link to `[[my_page_20200411_20200412]]`?

**Comment by wbthomason** [comment]
Yeah, those are good considerations. I had envisioned this feature as not explicitly interacting with links - between the existing `wiki_link_target_map` feature and the existing page omnicompletion, I think there are already good options both for users who prefer writing out the full `[[my_page_20200411]] ` and those who prefer writing `[[my_page]]` and having `WikiLinkOpen` resolve the correct target for them.

To my mind, adding this feature to page creation resolves an asymmetry between the two main interfaces to page paths: page creation and page linking.

**Comment by lervag** [comment]
Would it suffice to ensure that `:WikiOpen` also uses `g:wiki_link_target_map`?

**Comment by lervag** [comment]
I've pushed an update now. Please let me know if this was what you wanted.

**Comment by wbthomason** [comment]
@lervag Thanks! I think this works for a sufficiently sophisticated `g:wiki_link_target_map` - for example, in the use case I give above, the function will need to check if the suffix is already present to avoid double-appends of the sort @jabirali mentions, the function will need to check if a page already exists (is being opened) or not (creating a new page), etc. 

Personally, I think it would still be cleaner to have separate variables for a `g:wiki_create_map` and `g:wiki_link_target_map`. I've poked around in the code a bit more and could probably put together a PR if you're open to this approach.

Incidentally, it looks like this change introduced a small bug - I submitted a fix in #66.
